src/CONFIG.js

```diff
@@ -49,11 +49,13 @@ export default {
     EXPENSIFY: {
         // Note: This will be EXACTLY what is set for EXPENSIFY_URL whether the proxy is enabled or not.
         EXPENSIFY_URL: expensifyURL,
-        SECURE_EXPENSIFY_URL: secureURLRoot,
         NEW_EXPENSIFY_URL: newExpensifyURL,
-        URL_API_ROOT: expensifyURLRoot,
-        STAGING_EXPENSIFY_URL: stagingExpensifyURL,
-        STAGING_SECURE_EXPENSIFY_URL: stagingSecureExpensifyUrl,
+
+        // The main API is usually PRODUCTION, but there are cases where it points to internal DEV server
+        PRIMARY_API_ROOT: expensifyURLRoot,
+        PRIMARY_SECURE_API_ROOT: secureURLRoot,
+        STAGING_API_ROOT: stagingExpensifyURL,
+        STAGING_SECURE_API_ROOT: stagingSecureExpensifyUrl,
```

Comment:
The reason for renaming `URL_API_ROOT`  to `PRIMARY_API_ROOT` is because the underlying value of `URL_API_ROOT` is (almost) always the production api, we want to make that clear and avoid confusion
(Though it's also possible point to an internal DEV url)

The API (on some envs) can be changed dynamically. `URL_API_ROOT` variable can be mistaken to always point to the correct API for the current ENV, so instead we make it clear by renaming it to `PRIMARY_API_ROOT` and introduce `ApiUtils.getApiRoot()` 

We have a "Staging toggle" that can switch between primary and staging 
It should be more clear that it switches between the two, and that it's not always between staging and prod

Reply:
>the underlying value of URL_API_ROOT is (almost) always the production api, we want to make that clear and avoid confusion

The url is based on the environment variables so maybe we can be more explicit to say `ENVIRONMENT_API_ROOT`? Something about `PRIMARY` makes me wonder if there are "other additional APIs" or something to worry about.

Reply:
> Something about PRIMARY makes me wonder if there are "other additional APIs" or something to worry about.

That's what I'm trying to do here - prevent people from blindly using it

Problem: "Developers don't know the API root can change dynamically and that the change would not be reflected in `URL_API_ROOT`" 
Instead they should use `getApiRoot()`

> The url is based on the environment variables so maybe we can be more explicit to say ENVIRONMENT_API_ROOT?

That sounds good, but it's a bit ambiguous to me - I don't make the connection that it's about env variables, but I think it's about the environment name e.g. staging | prod | dev  
So I would think that on staging ENVIRONMENT_API_ROOT is `staging.exepsnify.com`

And this leads to Problem 2: "Developers think `URL_API_ROOT` always refers to the build environment"
I initially renamed `URL_API_ROOT` to `PROD_API_ROOT`, to make that clear
- on DEV (external contributors) it would referer to `www.expensify`)
- on PROD it refer to `www.expensify`
- on STAGING **it would again refer to `www.expensify`** (see .env.staging) 

Some developers (me included) used to think that, on staging `URL_API_ROOT` would be `staging.expensify` when it's actually `www.expensify` 

This is an inconsistency problem which remained hidden for a very long time, because it didn't result in noticeable issues - when we're on staging we're allowed use the prod api and some requests (like attachments and images) would use it

The reason this happens is because some links are configured to use `URL_API_ROOT` which always points to PROD (unless overridden by internal developer)

This PR aims to expose such usages by renaming the constant

I selected the word PRIMARY, because of the "Staging toggle" in settings - it switches between PRIMARY and STAGING, where PRIMARY
- on PROD is prod
- on STAGING is **prod**
- on DEV it's either prod or env value (internal dev) 

---

This is not the main goal of the PR, I just wanted to make the rename and draw attention to usages, so maybe we should revert the renames and do a separate PR for the problems outlined here 

It might be best to declare all URLs - staging / dev / prod and sort out the logic in `ApiUtils`, so we don't have a constant like PRIMARY that might be PROD or internal DEV url

Reply:
It feels like we are fundamentally worried that people will use this when they shouldn't - that sounds great. I'm mostly trying to avoid the idea that there are different levels of API or implying importance. 

I get the feedback about environment though.. so any of these alternatives could work equally well...

`STATIC_API_ROOT`
`FIXED_API_ROOT`
`DEFAULT_API_ROOT`
`NON_DYNAMIC_API_ROOT`

?

I think the main thing here is that the URL we are calling PRIMARY - is the one that doesn't change.

Reply:
I'll update to `DEFAULT_API_ROOT` 

One confusing part that remains is `DEFAULT_API_ROOT` on staging would be `www.expensify`, so by default staging should not use `DEFAULT_API_ROOT` but `STAGING_API_ROOT` 
(The new code comment explains why that is so) 

What I'd like to be able to change in the future is to make the configuration in such a way that 
DEFAULT_API, always returns the default for the current environment (so it would return `staging.expensify` on staging)